resources/codegen/tests/test___init__.py

```diff
@@ -0,0 +1,318 @@
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_globals_supplied_then_generator_called_with_updated_globals(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        mock_language_a,
+        a_capturing_logger,
+    ):
+
+        extra_globals = {"a": 1, "b": "🤠", "copyrightOwner": "Me"}
+
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            template_globals=extra_globals,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals.update(extra_globals)
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_globals_supplied_include_language_then_generator_global_not_overridden(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        mock_language_a,
+        a_capturing_logger,
+    ):
+
+        extra_globals = {"language": "b"}
+
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=["a"],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            template_globals=extra_globals,
+        )
+
+        expected_globals = openassetio_codegen.generators.helpers.default_template_globals()
+        expected_globals["language"] = "a"
+
+        call_args = mock_language_a.generate.call_args[0]
+        assert call_args[1] == expected_globals
+
+    def test_when_dry_run_set_true_then_structure_is_logged_as_info(
+        self,
+        yaml_path_all,
+        some_output_dir,
+        some_creation_callback,
+        a_capturing_logger,
+        structure_all_log_messages,
+    ):
+        openassetio_codegen.generate(
+            description_path=yaml_path_all,
+            output_directory=some_output_dir,
+            languages=[],
+            creation_callback=some_creation_callback,
+            logger=a_capturing_logger,
+            dry_run=True,
+        )
+
+        assert a_capturing_logger.handlers[0].messages == structure_all_log_messages
+
+
+@pytest.fixture
+def mock_language_a():
+    mock_generator = mock.Mock()
+    mock_generator.generate = mock.Mock()
+    openassetio_codegen.generators.a = mock_generator
```

Comment:
[consider] Use `monkeypatch` here and in `mock_language_b` to ensure the namespace is left unpolluted.

Reply:
Great idea - fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/5fa2e3aed45f2b39aac15fffb604a36fd387a4e8).